catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -414,6 +442,64 @@ public void init() {
             return String.format("{\"%s\":\"%s\"}", "local-catalog-id", catalogFramework.getId());
         });
 
+        post("/transform/csv", APPLICATION_JSON, (req, res) -> {
+            CsvTransform queryTransform = mapper.readValue(req.body(), CsvTransform.class);
+            Map<String, Object> transformMap = mapper.parser()
+                    .parseMap(req.body());
+            queryTransform.setMetacards((List<Map<String, Object>>) transformMap.get("metacards"));
+
+            List<Result> metacards = queryTransform.getTransformedMetacards(types,
+                    attributeRegistry)
+                    .stream()
+                    .map(ResultImpl::new)
+                    .collect(Collectors.toList());
+
+            SourceResponseImpl response = new SourceResponseImpl(null,
+                    metacards,
+                    Long.valueOf(metacards.size()));
+
+            Map<String, Serializable> arguments = ImmutableMap.<String, Serializable>builder().put(
+                    "hiddenFields",
+                    new HashSet<>(queryTransform.getHiddenFields()))
+                    .put("columnOrder", new ArrayList<>(queryTransform.getColumnOrder()))
+                    .put("aliases", new HashMap<>(queryTransform.getColumnAliasMap()))
+                    .build();
+
+            BinaryContent content = csvQueryResponseTransformer.transform(response, arguments);
+
+            String acceptEncoding = req.headers("Accept-Encoding");
+            // Very naive way to handle accept encoding, does not respect full spec
+            boolean shouldGzip =
+                    StringUtils.isNotBlank(acceptEncoding) && acceptEncoding.toLowerCase()
+                            .contains("gzip");
+
+            // Respond with content
+            res.type("text/csv");
+            String attachment = String.format("attachment;filename=export-%s.csv",
+                    Instant.now()
+                            .toString());
+            res.header("Content-Disposition", attachment);
+            if (shouldGzip) {
```

Comment:
See https://github.com/codice/ddf/pull/1760 for more details.

Typically, simply setting the content encoding relays to Spark that it should figure out the needing for compression, but there is actually a bug with how they do that (resulting in gzipped responses when they should not be). On top of that, the way to return non-text responses seems to be getting the raw ServletResponse and writing to the output stream which threw an exception every time when Spark did try to gzip it.